app/com/arpnetworking/play/configuration/ConfigurationHelper.java

```diff
@@ -119,12 +123,38 @@ public static FiniteDuration getFiniteDuration(final Config configuration, final
             final Injector injector,
             final Environment environment,
             final Config configuration) {
-        final Class<? extends T> clazz = getType(environment, configuration, "type");
+        final Class<? extends T> clazz = getType(environment, configuration, TYPE_KEY);
         return injector
                 .createChildInjector(new ConfigurationOverrideModule(configuration))
                 .getInstance(clazz);
     }
 
+    /**
+     * Use Guice to instantiate a POJO from configuration, deserializing directly from the
+     * configuration object.
+     *
+     * @param injector The injector to use to create the object.
+     * @param environment Play {@link Environment} instance.
+     * @param configuration The config to instantiate the object from. (Note that this should almost certainly <i>not</i> be
+     *   the entire Play {@link Config} instance; instead it is some sub-object describing the Java object to instantiate.
+     * @param <T> The type of object to instantiate.
+     * @return The instantiated object.
+     */
+    public static <T> T toInstanceMapped(
```

Comment:
This may or may not be necessary - I might be misunderstanding how Guice and Jackson interact.

I want to be able to deserialize a class defined outside of this repo from a configuration object. Since there will be no Guice provider available (because as mentioned, it's an external class), I want to be able to just read the constructor arguments from the configuration object as-is.

The most straightforward way I could think of was to turn the config object into JSON and then deserialize it via Jackson. Jackson will then just use the constructor parameter names, and there's no fussing around with any Hocon Config bindings.

e.g. `FileWatchingConfigProvider(final String path)` would be then configured in Hocon as
```
configProvider = {
    path = "/etc/config.json"
}
```

Reply:
It looks like we already do this manually elsewhere, like for per-type report configuration